### ISSUE TITLE
Code doesn't increase if run bundleRelease

### ISSUE THREAD

**Issue by yaroslav-shlapak** [davideas/GrabVer#18]
From the docs: 

> // - Code is increased if exists a task that contains: "bundle", "grabverRelease", ":module:assembleRelease"

But when I run ./gradlew bundleRelease - it doesn't increase
When I run 
```
./gradlew grabverRelease bundleRelease
```
 it increases but it is not being stored in the verstion.properties

The only way it works when I run 
```
./gradlew grabverRelease bundleRelease
```
with gradle configuration:
```
versioning {
    major = 1
    minor = 8
    dependsOn = "bundleRelease"
}
```
The logic in: https://github.com/davideas/GrabVer/blob/fd9b33e61c4d68d97910c000b53cec09ab0b5823/src/main/groovy/eu/davidea/gradle/GrabVer.groovy#L91-L93
seems to be a reason - there are no "bundle" there

**Comment by davideas** [comment]
Merged pull request #20.

**Comment by Kiran-B** [comment]
My issue is a mix of this issue and #19. I am trying to use this plugin on a Gradle based Spring Boot project. 

Though I see a console log `INFO - Saving versioning` the `version.properties` does not contain the latest version. And hence after each run, the project continues to have the same version. 
If `version.properties` has any content(say I put the PATCH version manually), it gets deleted and only a commented `timestamp` remains in the first line after the gradle execution.


My config is:
```groovy
versioning {
    // required (number)
    major 0
    minor 0
    // optional, force custom patch (number)
    patch 1
    // optional (any string)
    preRelease "SNAPSHOT"
    // optional, save new versioning on file on specified <task-name>
    // saveOn "bootDistZip"
}

```

When I execute the `./gradlew bootDistZip grabverRelease`, I get the following response:
```bash
> Configure project :
====== STARTED GrabVer v1.0.1
INFO - ProjectName 'foo'
INFO - Versioning root project 'foo'
INFO - Versioning filename '/Volumes/Projects/spring-svc-foo/version.properties'
INFO - Current versioning 0.0.0 #0 code=1
INFO - runTasks=[bootDistZip, grabverRelease]
INFO - Running with release build => 'Code' version will auto increment
INFO - Auto incrementing build number
INFO - Auto incrementing code version

> Task :compileJava UP-TO-DATE
INFO - foo:compileJava TaskState succeeded
INFO - Saving versioning 0.0.1-SNAPSHOT #1 code=2
====== ENDED GrabVer


Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/4.10.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 0s
5 actionable tasks: 5 up-to-date
```

**Comment by Kiran-B** [comment]
Just a note. Even if I add `saveOn "bootDistZip"`, the bug persists. That is, versions are not stored in `version.properties`.

**Comment by yaroslav-shlapak** [comment]
@Kiran-B could you please try 
`./gradlew grabverRelease bootDistZip`
the order matter with this plugin, though it is not obvoius

**Comment by Kiran-B** [comment]
@yaroslav-shlapak No it did not work for me :(

I still get an empty `version.properties` with just the latest timestamp in the first line. Whats weird is that the log has the entry
```bash
INFO - Saving versioning 0.0.1-SNAPSHOT #1 code=2
```

The above log I believe is from: https://github.com/davideas/GrabVer/blob/ead50fed01d12d74032ce4bacffaa41957e89246/src/main/groovy/eu/davidea/gradle/GrabVer.groovy#L100

which means this line should have executed saving the file correctly:
https://github.com/davideas/GrabVer/blob/ead50fed01d12d74032ce4bacffaa41957e89246/src/main/groovy/eu/davidea/gradle/GrabVer.groovy#L108

**Comment by davideas** [comment]
As I now understood the Android bundle build has the bundleVariant as: `bundleDebug` and `bundleRelease` not simply as `bundle`: https://developer.android.com/studio/build/building-cmdline#build_bundle.
I will add these task names in the next version, but I can't test because I have no projects with bundle configuration.

**Comment by Kiran-B** [comment]
I managed to workaround this, my apologies for not noting it here

Two things:

1. My first mistake: (TLDR version) I had preRelease set.
Details:
My config had the following:
```json
versioning {
    preRelease "SNAPSHOT"
}
```
As per this https://github.com/davideas/GrabVer/blob/ead50fed01d12d74032ce4bacffaa41957e89246/src/main/groovy/eu/davidea/gradle/VersioningExtension.groovy#L52-L53

if the preRelease is set, it would skip incrementing the PATCH (or more accurately it would reset PATCH to zero).

My bad: I had missed out this section in Rules in REadMe :

> patch - If not specified by user, increases at each release, but it auto resets back to 0 when Minor or Major version changes or **if preRelease is set**.


The fix for me was to remove `preRelease`.

2. My second mistake: (TLDR version) I used Java 11
Details:
Using this plugin with Java 11 results in an empty file `version.properties` file. Well not empty, but with just the latest timestamp in the first line.

The fix for me was to use Java 1.8.
